synapse/storage/databases/main/events_worker.py

```diff
@@ -1056,7 +1070,11 @@ async def have_seen_events(self, event_ids):
             set[str]: The events we have already seen.
         """
         # if the event cache contains the event, obviously we've seen it.
-        results = {x for x in event_ids if self._get_event_cache.contains(x)}
+        results = {
+            x
+            for x in event_ids
+            if self._get_event_cache.contains((x,)) or x in self._in_memory_events
```

Comment:
The change from `_get_event_cache.contains(x)` to `_get_event_cache.contains((x,))` is a bug fix I think, since the get event cache uses `(event_id,)` as its keys.